component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -54,77 +102,215 @@ public static String getEncodingMethod() throws TOTPException {
+                if (authConfigNode.getNodeType() == Node.ELEMENT_NODE) {
+                    Element authConfigElement = (Element) authConfigNode;
+                    String AuthConfig = authConfigElement.getAttribute("name");
+                    if (AuthConfig.equals("totp")) {
+                        NodeList AuthConfigChildList = authConfigElement.getChildNodes();
+                        for (int j = 0; j < AuthConfigChildList.getLength(); j++) {
+                            Node authConfigChildNode = AuthConfigChildList.item(j);
+                            if (authConfigChildNode.getNodeType() == Node.ELEMENT_NODE) {
+                                Element authConfigChildElement = (Element) authConfigChildNode;
+                                String tagAttribute = AuthConfigChildList.item(j).getAttributes().getNamedItem("name").getNodeValue();
+                                if (tagAttribute.equals("encodingMethod")) {
+                                    encodingMethod = authConfigChildElement.getTextContent();
+                                } else if (tagAttribute.equals("timeStepSize")) {
+                                    timeStepSize = authConfigChildElement.getTextContent();
+                                } else if (tagAttribute.equals("windowSize")) {
+                                    windowSize = authConfigChildElement.getTextContent();
+                                } else if (tagAttribute.equals("enableTOTP")) {
+                                    enableTOTP = authConfigChildElement.getTextContent();
+                                } else if (tagAttribute.equals("usecase")) {
+                                    usecase = authConfigChildElement.getTextContent();
+                                } else if (tagAttribute.equals("userAttribute")) {
+                                    userAttribute = authConfigChildElement.getTextContent();
+                                } else {
+                                    secondaryUserstore = authConfigChildElement.getTextContent();
+                                }
+                            }
+                        }
+                        break;
+                    }
+                }
+            }
+            if (context != null) {
+                context.setProperty("encodingMethod", encodingMethod);
+                context.setProperty("timeStepSize", timeStepSize);
+                context.setProperty("windowSize", windowSize);
+                context.setProperty("enableTOTP", enableTOTP);
+                context.setProperty("usecase", usecase);
+                context.setProperty("userAttribute", userAttribute);
+                context.setProperty("secondaryUserstore", secondaryUserstore);
+            }
+        } catch (SAXException | ParserConfigurationException | IOException e) {
+            throw new TOTPException("Cannot get the TOTP parameter values. ", e);
+        } catch (RegistryException e) {
```

Comment:
If there is no entry in the registry, It will check from local file.Since all fields are not mandatory we can have null values for them. For some parameters we have default values